File: Api/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace VirtaApi.Models
{
    public class Product
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();
        [Required]
        public string Title { get; set; }
        public decimal Price { get; set; }
        public string Description { get; set; }
        public virtual ICollection<ProductAttributes> Attributes { get; set; }
        public string Images { get; set; }
        public virtual ICollection<Category> Categories { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime UpdatedAt { get; set; } = DateTime.Now;
    }
}

File: Frontend/src/app/_models/product.ts
export interface Product {
    id: string;
    title: string;
    price: number;
    images: ProductImage[];
    description?: string;
    attributes?: ProductAttribute[];
}

export interface ProductImage {
    id: number;
    primary: boolean;
    url: string;
}

export interface ProductAttribute {
    id: number;
    title: string;
    value: string;
    priority: number;
}

export interface ProductInCart {
    id: string;
    title: string;
    price: number;
    images: string[];
    quantity: number;
}

export interface ProductInWishlist {
    id: string;
    title: string;
    price: number;
    images: string[];
}
